src/components/com_tjcertificate/administrator/controllers/bulktrainingrecord.json.php

```diff
@@ -0,0 +1,152 @@
+ */
+class TjCertificateControllerBulkTrainingRecord extends FormController
+{
+	protected $comMultiAgency = 'com_multiagency';
+
+	/**
+	 * Function to add multiple records
+	 *
+	 * @return  void
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	public function addRecords()
+	{
+		$app = Factory::getApplication();
+
+		if (!Session::checkToken())
+		{
+			$app->enqueueMessage(Text::_('JINVALID_TOKEN'), 'error');
+			echo new JsonResponse(null, null, true);
+			$app->close();
+		}
+
+		$user = Factory::getUser();
+		$data = $app->input->post->get('jform', array(), 'array');
+
+		$model = $this->getModel();
+
+		// Validate the posted data.
+		$form = $model->getForm($data, false);
+		$data = $model->validate($form, $data);
+
+		if ($data == false)
+		{
+			$errors = $model->getErrors();
+
+			if (!empty($errors))
+			{
+				$msg  = array();
+
+				// Push up to three validation messages out to the user.
+				for ($i = 0, $n = count($errors); $i < $n && $i < 3; $i++)
+				{
+					if ($errors[$i] instanceof Exception)
+					{
+						$msg[] = $errors[$i]->getMessage();
+					}
+					else
+					{
+						$msg[] = $errors[$i];
+					}
+				}
+
+				$errormsg = implode("<br>", $msg);
+				echo new JsonResponse(0, $errormsg, true);
+			}
+		}
+		else
+		{
+			$params              = ComponentHelper::getParams('com_tjcertificate');
+			$userIds             = $data['assigned_user_id'];
+			$data['client']      = "external";
+			$data['is_external'] = 1;
+
+			unset($data['assigned_user_id']);
+
+			foreach ($userIds as $userId)
+			{
+				if (ComponentHelper::isEnabled($this->comMultiAgency) && $params->get('enable_multiagency'))
+				{
+					$manageOwn = $user->authorise('core.manage.own.agency.user', $this->comMultiAgency);
+					$manage    = $user->authorise('core.manage.all.agency.user', $this->comMultiAgency);
+
+					if ($manageOwn && empty($manage))
+					{
+						$agencyModel = TJCERT::model('Agency', array('ignore_request' => true));
+
+						// Get agencies of logged-in user and assigned user
+						$assignedUserAgencies = $agencyModel->getUserAgencies($userId);
+						$loggedInUserAgencies = $agencyModel->getUserAgencies($user->id);
+
+						$result = '';
+
+						if (!empty($assignedUserAgencies && $loggedInUserAgencies))
+						{
+							// Convert object to array
+							$assignedUserAgencyArr = array_column($assignedUserAgencies, 'id');
+							$loggedInUserAgencyArr = array_column($loggedInUserAgencies, 'id');
+
+							// Compare both users agencies
+							$result = array_intersect($loggedInUserAgencyArr, $assignedUserAgencyArr);
+						}
+
+						if (empty($result))
+						{
+							continue;
+						}
+					}
+				}
+
+				$data['user_id'] = $userId;
+				$returnData = array();
+
+				if (ComponentHelper::isEnabled('com_tjqueue') && $params->get('tjqueue_records'))
+				{
+					$recordsModel      = TJCERT::model('BulkTrainingRecord', array('ignore_request' => true));
+					$response          = $recordsModel->queueRecords($data);
```

Comment:
change function name to addToQueue($data)

Reply:
updated